File: advent_of_code/2022/day1/src/main.rs
use std::{
    fs::{read_to_string, File},
    io::{BufReader, Read},
    path::Path,
};

mod part1;
mod part2;
fn main() {
    let input = read_to_string("src/input.txt").expect("Failed to read the file");

    let ans = part1::solve(&input);
    println!("Ans: {}", ans);

    let ans = part2::solve(&input);
    println!("Ans: {:?}", ans);
}

File: advent_of_code/2022/day1/zig/src/main.zig
const std = @import("std");
const part1 = @import("part1.zig");
const part2 = @import("part2.zig");

pub fn main() !void {
    const input_file = try std.fs.cwd().openFile("src/input.txt", .{});
    defer input_file.close();

    var result: i32 = 0;
    const input = try input_file.readToEndAlloc(std.heap.page_allocator, 1024 * 1024);
    result = try part1.solve(input);

    std.debug.print("Part1: {!}\n", .{result});

    result = try part2.solve(input);

    std.debug.print("Part2: {!}\n", .{result});
}
